ray_on_golem/client/client.py

```diff
@@ -192,3 +203,62 @@ def _make_request(
             raise RayOnGolemClientValidationError(
                 "Couldn't validate response data",
             ) from e
+
+    def _start_webserver(self) -> None:
+        with cli_logger.group("Ray On Golem webserver"):
+            if self._is_webserver_running():
+                cli_logger.print("Webserver is already running")
+                return
+
+            cli_logger.print("Starting webserver...")
+            subprocess.Popen(
+                [RAY_ON_GOLEM_PATH, "-p", str(self._base_url.port), "--self-shutdown"],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                start_new_session=True,
+            )
+
+            for _ in range(10):
```

Comment:
nitpicking but possibly such magic values as the number of retries and the interval between them should be extracted to module-level constants ...

Reply:
Good point, added configurable and time based waiting.